activitysim/core/workflow/tracing.py

```diff
@@ -0,0 +1,741 @@
+
+        if len(trace_results.index) == 0:
+            return
+
+        # write out the raw dataframe
+
+        file_path = self._obj.filesystem.get_trace_file_path(
+            "%s.raw.csv" % label, tail=self.run_id
+        )
+        trace_results.to_csv(file_path, mode="a", index=True, header=True)
+
+        # if there are multiple targets, we want them in separate tables for readability
+        for target in targets:
+            df_target = trace_results[trace_results[slicer_column_name] == target]
+
+            # we want the transposed columns in predictable order
+            df_target.sort_index(inplace=True)
+
+            # # remove the slicer (person_id or hh_id) column?
+            # del df_target[slicer_column_name]
+
+            target_label = "%s.%s.%s" % (label, slicer_column_name, target)
+
+            self.trace_df(
+                df_target,
+                label=target_label,
+                slicer="NONE",
+                transpose=True,
+                column_labels=["expression", None],
+                warn_if_empty=False,
+            )
+
+    def interaction_trace_rows(self, interaction_df, choosers, sample_size=None):
+        """
+        Trace model design for interaction_simulate
+
+        Parameters
+        ----------
+        interaction_df: pandas.DataFrame
+            traced model_design dataframe
+        choosers: pandas.DataFrame
+            interaction_simulate choosers
+            (needed to filter the model_design dataframe by traced hh or person id)
+        sample_size int or None
+            int for constant sample size, or None if choosers have different numbers of alternatives
+        Returns
+        -------
+        trace_rows : numpy.ndarray
+            array of booleans to flag which rows in interaction_df to trace
+
+        trace_ids : tuple (str,  numpy.ndarray)
+            column name and array of trace_ids mapping trace_rows to their target_id
+            for use by trace_interaction_eval_results which needs to know target_id
+            so it can create separate tables for each distinct target for readability
+        """
+
+        # slicer column name and id targets to use for chooser id added to model_design dataframe
+        # currently we only ever slice by person_id, but that could change, so we check here...
+
+        traceable_table_ids = self.traceable_table_ids
+
+        # Determine whether actual tables or proto_ tables for disaggregate accessibilities
+        persons_table_name = set(traceable_table_ids).intersection(
+            ["persons", "proto_persons"]
+        )
+        households_table_name = set(traceable_table_ids).intersection(
+            ["households", "proto_households"]
+        )
+
+        assert len(persons_table_name) == 1 and len(persons_table_name) == 1
+        persons_table_name, households_table_name = (
+            persons_table_name.pop(),
+            households_table_name.pop(),
+        )
+
+        if (
```

Comment:
This needs to be updated to trace the proto_population in disaggregate accessibility.  Trying to trace right now crashes.  You can just copy over the code from [here](https://github.com/SANDAG/activitysim/blob/BayDAG_estimation/activitysim/core/tracing.py#L779) (lines 779 to 805) and it seems to work fine.